modeshape-jcr/src/main/java/org/modeshape/jcr/JcrVersionManager.java

```diff
@@ -374,8 +374,9 @@ JcrVersionNode checkin( AbstractJcrNode node ) throws RepositoryException {
         } finally {
             // TODO: Versioning: may want to catch this block and retry, if the new version name couldn't be created
         }
-
-        return (JcrVersionNode)session.node(version, Type.VERSION);
+        // return the version node from the system cache, not this session's cache because this session is asynchronously 
+        // notified of system ws changes and may not always get the latest data immediately
+        return (JcrVersionNode)session.node(version, systemSession, Type.VERSION, null);
```

Comment:
I know this relates to versioning, but might we miss out on changes that we've made and haven't yet committed? Or do we always want the version manager to read the data from the system session.